PythonVisualizations/Array.py

```diff
@@ -258,9 +258,54 @@ def remove(self, val):
             return True
         return False
 
+    def traverse(self):
+        self.cleanUp()
+
+        # draw an index pointing to the first cell
+        indexDisplay = self.createIndex(0, 'j')
+        self.cleanup |= set(indexDisplay)
+
+        # draw output box
+        canvasDimensions = self.widgetDimensions(self.canvas)
+        outputBox = self.canvas.create_rectangle(\
+            ARRAY_X0+ CELL_SIZE//2, ARRAY_Y0 + canvasDimensions[1] // 3, ARRAY_X0 + CELL_SIZE*len(self.list), \
+            ARRAY_Y0 + canvasDimensions[1] // 3 + 50,\
+            fill = OPERATIONS_BG)

```

Comment:
I'd like to center the output box on the canvas.  That would help make it seem a little more disconnected with the data structure and more connected to the interface.  
Also you don't need to put backslash at the end of lines when they are part of a expression enclosed in brackets or parentheses.  Here's a suggested sizing for the outputBox:
```
        spacing = CELL_SIZE * 3 // 4
        padding = 10
        outputBox = self.canvas.create_rectangle(
            (canvasDimensions[0] - len(self.list) * spacing - padding) // 2,
            canvasDimensions[1] - CELL_SIZE - padding,
            (canvasDimensions[0] + len(self.list) * spacing + padding) // 2,
            canvasDimensions[1] - padding,
            fill = OPERATIONS_BG)
```